prov/efa/src/rdm/efa_rdm_pke_cmd.c

```diff
@@ -961,7 +961,13 @@ void efa_rdm_pke_handle_recv_completion(struct efa_rdm_pke *pkt_entry)
 		zcpy_rxe = pkt_entry->ope;
 	}
 
-	efa_rdm_pke_proc_received(pkt_entry);
+	if (ep->use_zcpy_rx) {
```

Comment:
This code block should be moved to the `if (pkt_entry->alloc_type == EFA_RDM_PKE_FROM_USER_BUFFER) {` above. When pkt entry is allocated from user buffer, we do the zcpy recv handling, otherwise we do the non-zcpy recv handling via `efa_rdm_pke_proc_received`.

But Brian is doubting if this shortcut is the best approach, we can discuss that